test/test_gaussian.py

```diff
@@ -297,6 +297,41 @@ def test_eager_subs_variable():
     assert g2.precision is g1.precision
 
 
+@pytest.mark.parametrize('subs', [
+    (('x', 'Variable("u", reals()) * 2'),),
+    (('y', 'Variable("v", reals(4)) + 1'),),
+    (('z', 'Variable("w", reals(6)).reshape((2,3))'),),
+    (('x', 'Variable("v", reals(4)).sum()'),
+     ('y', 'Variable("v", reals(4)) - 1')),
+    (('x', 'Variable("u", reals()) * 2 + 1'),
+     ('y', 'Variable("u", reals()) * Tensor(torch.ones(4))'),
+     ('z', 'Variable("u", reals()) * Tensor(torch.ones(2, 3))')),
+])
+@pytest.mark.parametrize('g_ints', ["", "i", "j", "ij"])
+@pytest.mark.parametrize('subs_ints', ["", "i", "j", "ji"])
+def test_eager_subs_affine(subs, g_ints, subs_ints):
+    sizes = {'i': 5, 'j': 6}
+    subs_inputs = OrderedDict((k, bint(sizes[k])) for k in subs_ints)
+    g_inputs = OrderedDict((k, bint(sizes[k])) for k in g_ints)
+    g_inputs['x'] = reals()
+    g_inputs['y'] = reals(4)
+    g_inputs['z'] = reals(2, 3)
+    g = random_gaussian(g_inputs)
+    subs = {k: eval(v) + random_tensor(subs_inputs) for k, v in subs}
+
+    inputs = g.inputs.copy()
+    for v in subs.values():
+        inputs.update(v.inputs)
+    grounding_subs = {k: random_tensor(OrderedDict(), d) for k, d in inputs.items()}
+    ground_subs = {k: v(**grounding_subs) for k, v in subs.items()}
+
+    g_subs = g(**subs)
+    assert issubclass(type(g_subs), GaussianMixture)
+    actual = g_subs(**grounding_subs)
+    expected = g(**ground_subs)(**grounding_subs)
+    assert_close(actual, expected, atol=1e-3, rtol=1e-4)
```

Comment:
Finally, I can understand the purpose of each statement in this test. :D I will review the logic `eager_subs_affine` soon.